application/preprocessing/generate_features.py

```diff
@@ -0,0 +1,190 @@
+import pandas as pd
+import sys, os
+import numpy
+import json
+from ..util import top_words
+from ..util import nltk_helper
+import nltk
+from collections import Counter
+
+# helper function to parse lyrics string in the csv
+def __get_lyrics(song):
+    lyrics_str = song['lyrics']
+
+    # Split into the word-count pairs, and split those pairs into an array as well and convert strings to numbers
+    # e.g. [[1,2], [3,1]] is two occurences of the #1 word and one occurence of the #3 word
+    lyrics = [[int(y) for y in x.split(':')] for x in lyrics_str.split()]
+
+    # finally convert to zero based so we can index an array
+    return [(x[0]-1, x[1]) for x in lyrics]
```

Comment:
Based on the issue I just ran into, you might want to flip the return order here to avoid the same issue I did, where I got my parameter order reversed.